source/reduceBy.js

```diff
@@ -50,7 +51,7 @@ var reduceBy = _curryN(4, [], _dispatchable([], _xreduceBy,
   function reduceBy(valueFn, valueAcc, keyFn, list) {
     return _reduce(function(acc, elt) {
       var key = keyFn(elt);
-      acc[key] = valueFn(_has(key, acc) ? acc[key] : valueAcc, elt);
+      acc[key] = valueFn(_has(key, acc) ? acc[key] : _clone(valueAcc, [], [], false), elt);
```

Comment:
Here I shallowly copy `valueAcc`
However, it can be `deep`

Reply:
I don't think it's worth it to do a deep clone here. Shallow clone is enough to eliminate the 'bug' and if a user is doing deep mutation of their accumulator I tend to feel like that's on them...